branchpro/tests/test_new_posterior.py

```diff
@@ -0,0 +1,1218 @@
+
+#
+# Test PoissonBranchProLogLik Class
+#
+
+
+class TestPoissonBranchProLogLik(unittest.TestCase):
+    """
+    Test the 'PoissonBranchProLogLik' class.
+    """
+    def test__init__(self):
+        df = pd.DataFrame({
+            'Time': [1, 2, 3, 5, 6],
+            'Incidence Number': [10, 3, 4, 6, 9]
+        })
+        ser_int = [1, 2]
+
+        bp.PoissonBranchProLogLik(df, ser_int, 6)
+
+        with self.assertRaises(TypeError) as test_excep:
+            bp.PoissonBranchProLogLik('0', ser_int, 2)
+        self.assertTrue('Incidence data has to' in str(test_excep.exception))
+
+        with self.assertRaises(TypeError) as test_excep:
+            bp.PoissonBranchProLogLik(df, 0, 2)
+        self.assertTrue('must be iterable' in str(test_excep.exception))
+
+        with self.assertRaises(TypeError) as test_excep:
+            bp.PoissonBranchProLogLik(df, ['zero'], 2)
+        self.assertTrue(
+            'distribution must contain' in str(test_excep.exception))
+
+        with self.assertRaises(ValueError) as test_excep:
+            bp.PoissonBranchProLogLik(df, ser_int, 2, time_key='t')
+        self.assertTrue('No time column' in str(test_excep.exception))
+
+        with self.assertRaises(ValueError) as test_excep:
+            bp.PoissonBranchProLogLik(df, ser_int, 2, inc_key='i')
+        self.assertTrue('No incidence column' in str(test_excep.exception))
+
+    def test_n_parameters(self):
+        df = pd.DataFrame({
+            'Time': [1, 2, 3, 5, 6],
+            'Incidence Number': [10, 3, 4, 6, 9]
+        })
+        ser_int = [1, 2]
+
+        log_lik = bp.PoissonBranchProLogLik(df, ser_int, 3)
+
+        self.assertEqual(log_lik.n_parameters(), 2)
+
+    def test_get_serial_intervals(self):
+        df = pd.DataFrame({
+            'Time': [1, 2, 3, 5, 6],
+            'Incidence Number': [10, 3, 4, 6, 9]
+        })
+        ser_int = [1, 2]
+
+        log_lik = bp.PoissonBranchProLogLik(df, ser_int, 3)
+        npt.assert_array_equal(
+            log_lik.get_serial_intervals(), np.array([1, 2]))
+
+    def test_set_serial_intervals(self):
+        df = pd.DataFrame({
+            'Time': [1, 2, 3, 5, 6],
+            'Incidence Number': [10, 3, 4, 6, 9]
+        })
+        ser_int = [1, 2]
+        new_ser_int = [1, 2, 1]
+        wrong_ser_int = (1)
+
+        log_lik = bp.PoissonBranchProLogLik(df, ser_int, 3)
+        log_lik.set_serial_intervals(new_ser_int)
+
+        npt.assert_array_equal(
+            log_lik.get_serial_intervals(), np.array([1, 2, 1]))
+
+        with self.assertRaises(ValueError):
+            log_lik.set_serial_intervals(wrong_ser_int)
+
+    def test__call__(self):
+        df = pd.DataFrame({
+            'Time': [1, 2, 3, 5, 6],
+            'Incidence Number': [0, 0, 0, 0, 0]
+        })
+        ser_int = [1, 2]
+
+        log_lik = bp.PoissonBranchProLogLik(df, ser_int, 3)
+
+        self.assertEqual(log_lik([1, 1]), 0)
+
+    def test_evaluateS1(self):
```

Comment:
For all the evaluate S1 tests, I would add a check that the derivative is close to what you get with a finite differences approximation of the derivative of the __call_ method